### PR TITLE
Added testing to rendered-target

### PR DESCRIPTION
**Pull request by SillyInventor**
Added a fake renderer with known predictable output
Added testing for most if not all functions

### Resolves

Partially addresses #471

### Proposed Changes

Add unit testing to rendered-target

### Reason for Changes

Everybody likes testing

### Test Coverage

Yup.

### CODE REVIEW

**Comment by thisandagain** [comment]
Sorry for the slow turnaround on this PR @SillyInventor. This looks great!

File: test/unit/sprites_rendered-target.js [hunk 1]
```diff
@@ -1,6 +1,66 @@
 var test = require('tap').test;
 var RenderedTarget = require('../../src/sprites/rendered-target');
 var Sprite = require('../../src/sprites/sprite');
+var Runtime = require('../../src/engine/runtime');
+
+
+var fakeRenderer = function (){
```
Comment (thisandagain) [inline test/unit/sprites_rendered-target.js]: Inconsistent spacing after `()`.

File: test/unit/sprites_rendered-target.js [hunk 2]
```diff
@@ -1,6 +1,66 @@
 var test = require('tap').test;
 var RenderedTarget = require('../../src/sprites/rendered-target');
 var Sprite = require('../../src/sprites/sprite');
+var Runtime = require('../../src/engine/runtime');
+
+
```
Comment (thisandagain) [inline test/unit/sprites_rendered-target.js]: Extra line break

File: test/unit/sprites_rendered-target.js (hunk 1 above)
Comment (thisandagain) [inline test/unit/sprites_rendered-target.js]: I feel like the "fakeRenderer" might be better split out and added as a module to "test/fixtures". Thoughts?

**Review by thisandagain — CHANGES_REQUESTED** [review]
This is great. Really minor comments / nitpicks.

**Comment by SillyInventor** [comment]
It is I who should apologize for the timing of the commit. I've been very busy testing tutorial layouts with students etc. and just sitting on this rather than committing. Hopefully the edits are good.

**Comment by thisandagain** [comment]
Looks great @SillyInventor. Thanks for splitting that out. Just for documentation:

## Before
image

## After
image

**Review by thisandagain — APPROVED** [review]

**Pull request merged**